dbt_platform_helper/providers/yaml_file.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+import yaml
+from yaml.parser import ParserError
+from yamllint import linter
+from yamllint.config import YamlLintConfig
+
+
+class YamlFileProviderException(Exception):
```

Comment:
Going with generalising File system errors what do you think of having the root exception being `FileProviderException` ?